oxidation/libparsec/crates/serialization_format/src/parser/protocol.rs

```diff
@@ -311,17 +333,24 @@ impl Cmds {
                         }
 
                         pub fn load(buf: &[u8]) -> Result<Self, ::rmp_serde::decode::Error> {
-                            Ok(if let Ok(data) = ::rmp_serde::from_slice::<Self>(buf) {
+                            let data = ::rmp_serde::from_slice::<Self>(buf);
+
+                            if data.is_ok() {
                                 data
                             } else {
                                 // Due to how Serde handles variant discriminant, we cannot express unknown status as a default case in the main schema
                                 // Instead we have this additional deserialization attempt fallback
-                                let data = ::rmp_serde::from_slice::<UnknownStatus>(buf)?;
-                                Self::UnknownStatus {
-                                    _status: data.status,
-                                    reason: data.reason,
+                                let unknown_status = ::rmp_serde::from_slice::<UnknownStatus>(buf)?;
+
+                                match unknown_status.status.as_str() {
+                                    "ok" => data,
+                                    #valid_statutes
+                                    _ => Ok(Self::UnknownStatus {
```

Comment:
why not `Ok(unknwon_status)` ?

Reply:
What do you mean ?

Reply:
replace `Ok(...)` body by my stuff

Reply:
Not the same thing, `Self::UnknownStatus` is part of the `Enum`, `unknown_status` is not